### ISSUE TITLE
Uncatchable SSHExceptions

### ISSUE THREAD

**Issue by Dreamsorcerer** [paramiko/paramiko#2016]
My application often throws errors like these in the logs:
```
ERROR:paramiko.transport:Exception: Error reading SSH protocol banner
ERROR:paramiko.transport:Traceback (most recent call last):
ERROR:paramiko.transport:  File "/usr/local/lib/python3.8/dist-packages/paramiko/transport.py", line 2211, in _check_banner
ERROR:paramiko.transport:    buf = self.packetizer.readline(timeout)
ERROR:paramiko.transport:  File "/usr/local/lib/python3.8/dist-packages/paramiko/packet.py", line 380, in readline
ERROR:paramiko.transport:    buf += self._read_timeout(timeout)
ERROR:paramiko.transport:  File "/usr/local/lib/python3.8/dist-packages/paramiko/packet.py", line 622, in _read_timeout
ERROR:paramiko.transport:    raise socket.timeout()
ERROR:paramiko.transport:socket.timeout
ERROR:paramiko.transport:
ERROR:paramiko.transport:During handling of the above exception, another exception occurred:
ERROR:paramiko.transport:
ERROR:paramiko.transport:Traceback (most recent call last):
ERROR:paramiko.transport:  File "/usr/local/lib/python3.8/dist-packages/paramiko/transport.py", line 2039, in run
ERROR:paramiko.transport:    self._check_banner()
ERROR:paramiko.transport:  File "/usr/local/lib/python3.8/dist-packages/paramiko/transport.py", line 2215, in _check_banner
ERROR:paramiko.transport:    raise SSHException(
ERROR:paramiko.transport:paramiko.ssh_exception.SSHException: Error reading SSH protocol banner
ERROR:paramiko.transport:
```

The biggest issue is that the traceback doesn't seem to go back to my code at all. So, I see no way to catch these errors.

**Comment by mhemeryck** [comment]
We did notice the same issue in our application.

The issue we did find was that even while we were correctly handling the exception, we would still see the errors in our logs.
This is because paramiko has its own error logging -- your stack trace actually already indicates it: `paramiko.transport:` is the name of the logger.

Our fix was simply to ignore the specific logger `paramiko.transport` since there's nothing actionable on our application, except for dealing with the exception. How yo fix that in your application depends on your logging setup (e.g. remove a handler).
